### PR TITLE
Fix #12656 by setting the proper reporter for inlined methods

### LINKED ISSUE #12656: Wrapped typeCheckErrors is emitting compilation errors
The `typeCheckErrors` when used inside another definition does not trap the error messages. 
Affects munit.

## Compiler version

v3.0.0 (also tested on nightly)

## Minimized code

```Scala
  transparent inline def expectCompileError(
      inline code: String,
      expectedMsg: String
  ) =
    val errors = compiletime.testing.typeCheckErrors(code)
    assert(errors.head.message == expectedMsg)

  expectCompileError("""compiletime.error("some error")""", "some error")
```

## Output

```scala
error: some error
```

## Expectation
No compilation error (the error should be trapped)

### PR DESCRIPTION
**Pull request by soronpo**
Fixes #12656

### CODE REVIEW

File: tests/run/i12656.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+transparent inline def expectCompileError(
+  inline code: String,
+  expectedMsg: String
+) =
+  val errors = compiletime.testing.typeCheckErrors(code)
```
Comment (nicolasstucki) [inline tests/run/i12656.scala:5]: We should add a similar test using `typeChecks` to make sure that one works and avoid regressions.

**Comment by soronpo** [comment]
@nicolasstucki I get a pickling test diff:
```
***** before-pickling.txt
                    <
                      <[ : scala.compiletime.testing.Error]:scala.Array[scala.compiletime.testing.Error]>@
                        tests\run\i12656.scala<108..108>
***** AFTER-PICKLING.TXT
                    <
                      <[ : scala.compiletime.testing.Error]:Seq[scala.compiletime.testing.Error]>@
                        tests\run\i12656.scala<108..108>
*****
```
Not sure what to do with this.

**Comment by nicolasstucki** [comment]
You can add it to `compiler/test/dotc/run-test-pickling.blacklist`. It might be good to open another issue on this. It does require investigation.

File: compiler/test/dotc/run-test-pickling.blacklist [hunk 2]
```diff
@@ -36,4 +36,5 @@ zero-arity-case-class.scala
 i12194.scala
 i12753
 t6138
-t6138-2
\ No newline at end of file
+t6138-2
+i12656.scala
```
Comment (nicolasstucki) [inline compiler/test/dotc/run-test-pickling.blacklist:40]:
```suggestion
i12656.scala

```

**Review by nicolasstucki — APPROVED** [review]

**Comment by soronpo** [comment]
I've discovered that this issue hid another bug. It seems that when an implicit macro issues an error message, a non-transparent inline function wrapper fails to pick it up. If the function is transparent, then it works as expected:

```scala
trait MacroErr
object MacroErr:
  import scala.quoted.*
  inline given MacroErr = ${ macroErr }
  def macroErr(using Quotes): Expr[MacroErr] =
    import quotes.reflect.*
    '{ compiletime.error("some macro error") }
```

```scala
inline def expectCompileError(
  inline code: String,
  expectedMsg: String
) =
  val errors = compiletime.testing.typeCheckErrors(code)
  assert(errors.head.message == expectedMsg, (errors.head.message, expectedMsg))

@main def Test = expectCompileError("""summon[MacroErr]""", "some macro error")
```

```
Exception in thread "main" java.lang.AssertionError: assertion failed: (no implicit argument of type MacroErr was found for parameter x of method summon in object Predef.
I found:

    MacroErr.given_MacroErr

But given instance given_MacroErr in object MacroErr does not match type MacroErr.,some macro error)
```

What do you think @nicolasstucki, should I file this as a separate issue and we let this one be merged as-is, or should we investigate further why non-transparent inline method wrappers fail in this particular area?

**Comment by nicolasstucki** [comment]
File it as a separate issue.

**Comment by soronpo** [comment]
Can we merge this?

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**